boards/xtensa/esp32/common/src/esp32_tca9548a.c

```diff
@@ -0,0 +1,145 @@
+/****************************************************************************
+ * boards/xtensa/esp32/common/src/esp32_tca9548a.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <stdio.h>
+#include <debug.h>
+
+#include <nuttx/arch.h>
+#include <nuttx/i2c/i2c_master.h>
+#include <nuttx/i2c/tca9548a.h>
+
+#include "esp32_board_i2c.h"
+#include "esp32_i2c.h"
+#include "esp32_tca9548a.h"
+
+FAR struct tca9548a_dev_s *g_tca9448a_devs[8] =
+                                                {
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL,
+                                                  NULL
+                                                };
```

Comment:
```suggestion
FAR struct tca9548a_dev_s *g_tca9448a_devs[8];
```
Is zero initialized by a standard. Only relevant in case if `NULL` is not defined to zero.
Or remove extra spaces and make like:
```
{
  NULL,
  NULL,
...
};
```

Reply:
Not defining it to NULL will create a dangling pointer, please do a test if you want to confirm it:
```
#include <stdio.h>

struct dev_s
{
  int a;
  int b;
  char c;
};

int main(void)
{
  struct dev_s *devs[8];
  struct dev_s *devx[8]=
  {
	  NULL,
	  NULL,
	  NULL,
	  NULL,
	  NULL,
	  NULL,
	  NULL,
	  NULL,
  };

  if (devs[0] == NULL)
  {
    printf("devs[0] is NULL\n");
  }
  else
  {
    printf("devs[0] is NOT NULL\n");
  }

  if (devx[0] == NULL)
  {
    printf("devx[0] is NULL\n");
  }
  else
  {
    printf("devx[0] is NOT NULL\n");
  }

  return 0;
}
```

Reply:
hmm, if a move it to global it will be allocated on bss and will be zeroed by default. Anyway I think it is save to initialize as NULL to avoid erratic behavior of compiler.